drivers: dac: add AD56x1 devices

mickael868 opened the pull request:
The Analog Device AD5601, AD5611, and AD5621 devices are 8, 10 and 12 bits DAC respectively.

These devices use a 16 bits SPI communication protocol. The 2 first bit encode the low power mode (this driver do not use the low power mode, it always run in the 'normal' mode). The next 14 bits contain the left aligned value. The 2, 4, or 6 remaining bits (depending on the resolution) are set to 0.

Notes:

1. A driver named `ad56xx` already exists but it is not compatible with these devices. From what I saw, the ad56xx driver does not cover any `AD56[0-9]1` device.
2. The Texas Instrument DAC8311 device is pin-to-pin and functionally compatible with this implementation, but with a 14 bits resolution (no right padding) and its minimum power supply voltage is 1.8 V instead of 2.7 V for the AD26x1.
From an implementation point of view, the DAC8311 is more closely related to the AD56x1 devices than to the TI DAC8411 which is a 16-bits DAC and requires a 3-bytes payload.

dsseng commented on drivers/dac/dac_ad56x1.c:
```diff
@@ -0,0 +1,156 @@
+		.bus = SPI_DT_SPEC_INST_GET(                                                       \
+			index, SPI_OP_MODE_MASTER | SPI_MODE_CPHA | SPI_WORD_SET(8), 0),           \
+		.resolution = res};                                                                \
+	DEVICE_DT_INST_DEFINE(index, ad56x1_init, NULL, &data_##name##_##index,                    \
```
Maybe remove `ad56x1_data` and just pass NULL here instead of an empty struct?

dsseng commented on drivers/dac/dac_ad56x1.c:
```diff
@@ -0,0 +1,156 @@
+
+	LOG_DBG("sending to DAC %s command 0x%02X, (value 0x%04X, normal mode)", dev->name, command,
+		value);
+	int result = spi_write_dt(&config->bus, &tx);
```
Please place variable definitions on the top of the block, at least when possible

MaureenHelm reviewed (changes requested):
Need to address compliance check and add the driver to `tests/drivers/build_all/dac/app.overlay`

Commit:
drivers: dac: add AD56x1 devices

The Analog Device AD5601, AD5611, and AD5621 devices are 8, 10 and 12
bits DAC respectively.

These devices use a 16 bits SPI communication protocol.
The 2 first bit encode the low power mode (this driver do not use the
low power mode, it always run in the 'normal' mode).
The next 14 bits contain the left aligned value. The 2, 4, or 6
remaining bits (depending on the resolution) are set to 0.

Signed-off-by: Mickael Bosch <mickael.bosch@linux.com>

mickael868 commented on drivers/dac/dac_ad56x1.c:
```diff
@@ -0,0 +1,152 @@
+
+#define DT_DRV_COMPAT adi_ad5601
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+#define DAC_AD5601_RESOLUTION 8
```
Sorry for the delay, I'm working on another project.

I did not get it: are you suggesting that the user has to specify the resolution in the dts file or that this driver sets the resolution so that the user is able to call `DT_PROP` to retrieve the resolution in their code?
